traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
```

Comment:
Possibly make this a named tuple?

Reply:
namedtuple turns this into a sequence-like object. Unlike the observer, this is a public-facing object, so it would be difficult going back. Maybe better to wait until we really need to.

Reply:
I agree about avoiding namedtuple; it's an interface that presents multiple ways to do it, which often isn't what you want. It's useful as a transitional type when going from something that was previously a tuple to something closer to a class with attributes, but I don't think it's what we want here. A dataclass may be closer to what we want, but that's not available until Python 3.7.